packages/test-commerce-sdk-react/app/components/_app-config/index.tsx

```diff
@@ -73,12 +73,17 @@ AppConfig.restore = () => {}
 AppConfig.extraGetPropsArgs = () => {}
 AppConfig.freeze = () => {}
 
-// Recommended settings for PWA-Kit usage.
+const isServerSide = typeof window === 'undefined'
+// Recommended settings for PWA-Kit usages.
 const options = {
     queryClientConfig: {
         defaultOptions: {
             queries: {
-                retry: false
+                retry: false,
+                // Avoid double requests during hydration
+                staleTime: 5 * 1000,
+                // So that the final value for the `isLoading` state is correct, whenever a query has an error in SSR.
+                ...(isServerSide ? {retryOnMount: false} : {})
```

Comment:
How do we test these features? If the default options live in the user land.

Reply:
Is this something that we have to test? Would we not be testing react query at that point, which has already been done in their own tests I'm assuming.

Reply:
With regards to the `retryOnMount` is there a reason why we chose to conditionally set it to `false` just for server-side? Could we not set `retryOnMount: false` across the board for client and server for simplicity sake?

Reply:
Commented here along similar lines https://github.com/SalesforceCommerceCloud/pwa-kit/pull/791#discussion_r1003686061

Reply:
> Is this something that we have to test? Would we not be testing react query at that point...

@bendvc I mean, how do we test that, for example, the double requests are no longer happening? I want to make sure that it works as expected and that a future regression will be easily caught.

Another part of the picture: this is a performance feature of our SDK that we used to have with `getProps`. So if we do have the react-query config be in user land, I feel uncomfortable that it looks like we can't test the double requests.

Reply:
With respect to the double request.. thats no longer a problem that we are solving in the sdk.. it's in the user land and we are really indirectly solving the problem. E.g. using a positive value staleTime ensures that duplicate calls aren't made on the first request. So if you are asking are we testing that staleTime works.. I'd say don't bother testing that.

Reply:
> From react-query's point of view, we are overriding their default value. As a consequence, I feel like it would be more likely for the users to override the option to some other value (maybe reset it back to react-query's default). And thus undoing our work.

I don't think we are giving our partners the benefit of the doubt on being able to understand that this code runs on the server and on the client. 

If they set the value back and run into a problem, they either figure out that "hey this runs on the server and client, I made a mistake, let me use a conditional instead" .. or they cant figure it out and contact us, we tell them whats wrong, they see it was their fault but they are happy with our customer service and they are educated. And we are all winners. 

If we see this is happening a lot, we make a ticket for it and look at a solution at that point. For now tho, I would like to keep the code simple and give them the benifit of the doubt. If we want, we can put a comment above the entire option block saying that this is executed on server and client and to be sure to use conditionals where needed. But that in my eyes is already apparent if we are using a conditional in our user-land default configuration, or is in our docs (which I'm going to be creating).

Reply:
I want to agree with @bendvc here and say that we're being what I would call "over responsible" here if we assume that we're going to effectively convey the full nuance of what the implementation should look like for the developer partner. The React Query docs relevant to these decisions could simply be linked via code comments along with us simply stating `// we are setting sensible defaults, please read the React Query docs <link> for more info`

Which is what I understand Ben to be saying when he calls out that we don't want to solve for a problem we don't have. The back-and-forth (while I don't want to shut down or discourage active discussion and thoughtful consideration of our decisions) can sometimes be "churn" or overthinking / over engineering a decision by spending too much time discussing a hypothetical problem that we don't have strong evidence for the existence of.

I can represent lesser-familiarity with React Query (I'm new to this feature) and say that our documentation on this should first explain the problem we aim to solve (in this case, "improve app performance via isomorphic API call execution"). Once a partner developer buys into that, they are agreeing to understand SSR a bit more. If they haven't bothered to look up SSR at all, then they should not be doing anything aside from blindly adopting our recommendation for configuring it.